src/main/java/com/google/impactdashboard/servlets/GetProjectDataServlet.java

```diff
@@ -13,13 +17,25 @@
 @WebServlet("/get-project-data")
 public class GetProjectDataServlet extends HttpServlet {
 
+  private ProjectDataRetriever projectDataRetriever;
   /**
    * Method called by the frontend to get the data needed to graph a projects information.
    * @param request contains the project data is requested for.
    * @param response contains json representation of information to graph.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    throw new UnsupportedOperationException("Not Implemented");
+    if(projectDataRetriever == null) {
+      projectDataRetriever = ProjectDataRetriever.create();
+    }
```

Comment:
Could you use the `init` method from `HttpServlet` here instead?

Reply:
Yes I can it will make it a lot easier to read.